salt/kubernetes-minion/init.sls

```diff
@@ -3,6 +3,23 @@
 #######################
 include:
   - repositories
+  - cert
+
+# etcd needs proper certs on the minions too
+{% from 'cert/init.sls' import ip_addresses %}
+
+{% for _, interface_addresses in grains['ip4_interfaces'].items() %}
+  {% for interface_address in interface_addresses %}
+    {% do ip_addresses.append("IP: " + interface_address) %}
+  {% endfor %}
+{% endfor %}
+# add some extra names the API server could have
+{% do ip_addresses.append("DNS: " + grains['fqdn']) %}
```

Comment:
We are mixing names and ip addresses here again. Maybe we can rename the `ip_addresses` var to something like `subject_alt_names` or something like that (everywhere, if we are going to mix DNS and IP entries on our SAN cert request). It's a minor thing, but when debugging can lead to mistakes...